calour/heatmap/heatmap.py

```diff
@@ -395,7 +416,18 @@ def plot(exp, sample_color_bars=None, feature_color_bars=None,
         The width of the tree relative to the main heatmap (12 is identical size)
     title : str (optional)
         The title of the figure.
-    kwargs : dict, optional
+    barwidth : float
+        The width of the bars
+    barspace : float
+        The spacing between the bars.
+    sample_highlight_colors : pd.Series or dict
+        The colors of the sample categories indicated on the sample axis
+    feature_highlight_colors : pd.Series or dict
```

Comment:
could you make this only accept `dict`? for both sample and feature colors. I don't see why `pd.Series` will make this simpler. try to keep this already complicated func API as simple as possible.

Reply:
same for the _ax_color_bar function

Reply:
Fret not - they behave nearly identically

```
>>> import pandas as pd
>>> dd = {1:'a', 2:'b', 3:'b'}
>>> dp = pd.Series({1:'a', 2:'b', 3:'b'})
>>> dp[1]
'a'
>>> dd[1]
'a'
```

If it is ok, I would like to keep the docs the way they are, so that people know that they can also directly pass in `pd.Series`

Reply:
Then the data type for `colors` should be `Mapping`. I don't know why you specifically bring up `pd.Series`. It unnecessarily imposes restriction and can be misleading. Either use the most common data type, ie, `dict` or use the most general type `Mapping` in the doc.

Reply:
fineeeeeee I changed it to only have it use `dicts`.  

Although we are just nitpicking semantics at this point -- it really doesn't matter, as long as it is a mapping type. 

But I don't think it is useful to reference `Mapping` types, because no one with the exception of skill devs know what that is.

Reply:
it ONLY take you ONE sentence to explain. and You can list common types allowed like this:

````
colors : ``Mapping``
    xxxxxxx. ``Mapping`` include common types like ``dict``, ``pd.Series``, ``defaultdict``, etc.
````

it is just confusing to unreasonably just say ``pd.Series``. I'd like to hold the quality for every PR as much as possible, since the code quality always rots along the time. We wanna  slow this down.